### DISCUSSION TITLE
PICO-W with WIZNET Ethernet in AP mode

### DISCUSSION THREAD

**Discussion by ondrej1024** [micropython/micropython#14482] [RP2040 / Pico]
Hi all,

I have a PICO-W connected to a WIZNET Ethernet module. So basically I have 2 network interfaces, Wifi and Ethernet.
When I bring up the Wifi interface in AP mode, it enables a DHCP server on _both_ interfaces. While this is expected on the Wifi interface, I don't want this on the Ethernet interface.

Does anyone know how to avoid having the DHCP server being enabled on the Ethernet interface?

**Comment by victorallume** [comment]
Do you want to have both interfaces active at the same time? If not, you could disable the ethernet interface while you're running the AP. I'm not sure how micropython handles having multiple active network interfaces - it's not a common scenario, and probably wasn't considered when the network code was initially implemented.

**Reply by ondrej1024** [reply]
Thanks a lot. Your answer actually gave me a clue.

The Ethernet interface is the primary one and will be activated on start up. If it fails (no cable plugged in or no DHCP server available in the LAN) then the Wifi interface will be activated in AP mode. 
So far so good. But when I plug in the Ethernet cable later on, the Pico starts responding to DHCP requests from the LAN, which messes up my LAN. So the solution is probably to shut down the Ethernet interface properly after it has failed initial connection.

**Reply by victorallume** [reply]
Yeah, I'm currently working on a system that uses multiple network interfaces, and it can get messy if you don't shut down one interface before using another. 

Another option would be to use the non-LWIP wiznet driver, which uses a different socket implementation, which should not interfere with wifi at all, but I'm not sure if it's possible presently possible to compile that in while also using LWIP for wifi (which I think is required)

**Comment by bstin** [comment]
I'm trying to compile RPI_PICO_W with WizNet module enabled however I cannot seem to get it working. My make command is:

`make -j BOARD=RPI_PICO_W MICROPY_PY_NETWORK_WIZNET5K=5500`

This compile succeeds, but when I load the resulting .uf2 file there is no WIZNET5x being included in network module? What am I missing?

**Reply by bstin** [reply]
Replying to myself to help others if searching. Adding `set(MICROPY_PY_NETWORK_WIZNET5K W5500)` to `boards/RPI_PICO_W/mpconfigboard.cmake` seems to have worked. 

Will likely also need to modify `boards/RPI_PICO_W/mpconfigboard.h` to set various SPI pin definitions for WIZNET module.

**Reply by ondrej1024** [reply]
> Replying to myself to help others if searching. Adding `set(MICROPY_PY_NETWORK_WIZNET5K W5500)` to `boards/RPI_PICO_W/mpconfigboard.cmake` seems to have worked.
> 
> Will likely also need to modify `boards/RPI_PICO_W/mpconfigboard.h` to set various SPI pin definitions for WIZNET module.

Yes, basically you have to merge the two board definitions of [RPI_PICO_W](https://github.com/micropython/micropython/tree/master/ports/rp2/boards/RPI_PICO_W) and [W5500_EVB_PICO](https://github.com/micropython/micropython/tree/master/ports/rp2/boards/W5500_EVB_PICO).
